tests/functional/pv/pvc_clone/test_clone_deletion_after_max_cluster_space_utilization.py

```diff
@@ -0,0 +1,213 @@
+        logger.info(f"pvc size: {self.pvc_size}")
+
+        self.pvc_obj = pvc_factory(
+            interface=interface_type, size=self.pvc_size, status=constants.STATUS_BOUND
+        )
+        self.pod_obj = pod_factory(
+            interface=interface_type, pvc=self.pvc_obj, status=constants.STATUS_RUNNING
+        )
+
+        # Calculating the file size as 95% of the PVC size - in MB
+        self.filesize = f"{int(self.pvc_size * 1024 * 0.95)}M"
+
+        self.pod_obj.run_io(
+            size=self.filesize,
+            io_direction="write",
+            storage_type="fs",
+        )
+
+        self.pod_obj.get_fio_results()
+        logger.info(f"IO finished on pod {self.pod_obj.name}")
+
+    def verify_alerts_via_prometheus(self, expected_alerts, threading_lock):
+        """
+        Verify Alerts on prometheus
+
+        Args:
+            expected_alerts (list): list of alert names
+            threading_lock (threading.Rlock): Lock object to prevent simultaneous calls to 'oc'
+
+        Returns:
+            bool: True if expected_alerts exist, False otherwise
+
+        """
+        prometheus = PrometheusAPI(threading_lock=threading_lock)
+        logger.info("Logging of all prometheus alerts started")
+        alerts_response = prometheus.get(
+            "alerts", payload={"silenced": False, "inhibited": False}
+        )
+        actual_alerts = list()
+        for alert in alerts_response.json().get("data").get("alerts"):
+            actual_alerts.append(alert.get("labels").get("alertname"))
+            print("Actual Alerts:", actual_alerts)
+        for expected_alert in expected_alerts:
+            if expected_alert not in actual_alerts:
+                logger.error(
+                    f"{expected_alert} alert does not exist in alerts list."
+                    f"The actaul alerts: {actual_alerts}"
+                )
+                return False
+        return True
+
+    @skipif_external_mode
+    @magenta_squad
+    def test_clone_deletion_after_max_cluster_space_utilization(
+        self, interface_type, pvc_clone_factory, threading_lock
+    ):
+        """
+        Steps:
+            1. Have a cluster with OCP + ODF internal mode
+            2. Create cephfs + RBD pvcs with data
+            3. Create clones of pvcs till cluster full ratio.
+            4. Make the cluster out of full by increasing the full ratio.
+            5. After the cluster is out of full state and IOs started , Try to delete clones.
+            6. Clone deletion should be successful and should not give error messages.
+        """
+        # Creating the clones one by one and wait until they bound
+        self.timeout = 1800
+        logger.info(
+            f"Start creating {self.num_of_clones} clones on {interface_type} PVC of size {self.pvc_size} GB."
+        )
+        clones_list = []
+        for i in range(self.num_of_clones):
+            index = i + 1
```

Comment:
Is this needed? as you mention the creation of clones after it gets created it seems unneccessary

Reply:
It is for how many clones are created. So will get that info

Reply:
Storing i value in index variable just to log it later is not clean way to do it in my opinion. You can simply print i+1 instead

Reply:
done